llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1901,6 +1901,58 @@ struct DSEState {
     return true;
   }
 
+  // Check if there is a dominating condition, that implies that the value
+  // being stored in a ptr is already present in the ptr.
+  bool dominatingConditionImpliesValue(MemoryDef *Def) {
+    auto *StoreI = dyn_cast<StoreInst>(Def->getMemoryInst());
```

Comment:
```suggestion
    auto *StoreI = cast<StoreInst>(Def->getMemoryInst());
```
To indicate that the cast must succeed.